Tensile/Tests/unit/test_mergeLogic.py

```diff
@@ -0,0 +1,180 @@
+                 for i in solutionPool)
+
+@pytest.mark.parametrize("sizes, expected", [
+  (   trimmedSize, [[128,128,1,128],[512,512,1,512],[1024,1024,1,1024]]),
+  (notTrimmedSize, [[1024,1024,1,1024],[128,128,1,128]])
+  ])
+def test_fixSizeInconsistencies(sizes, expected):
+  data = yaml.load(sizes, yaml.SafeLoader)
+  data_ = fixSizeInconsistencies(data[0], "dummy")
+
+  for [size, [_,_]], expected_ in zip(data_[0], expected):
+    assert size == expected_
+  # print final yaml for visual inspection
+  # stream = io.StringIO("")
+  # yaml.safe_dump(data_, stream, default_flow_style=None)
+  # print(stream.getvalue())
+
+@pytest.mark.parametrize("input,expectedNumKernelRemoved", [(baseLogic, 1),
+                                                            (incLogic, 0)])
+def test_removeUnusedKernels(input, expectedNumKernelRemoved):
+  data = yaml.load(input, yaml.SafeLoader)
+  dataFiltered, numKernelRemoved = removeUnusedKernels(data)
+
+  # test if number of solution removed is correct
+  assert numKernelRemoved == expectedNumKernelRemoved
+
+  # test if solution is re-indexed
+  for index, s in enumerate(dataFiltered[5]):
+    assert index == s["SolutionIndex"]
+
+  # check if size-kernel mapping is not compromised
+  solutionMapBefore = {}
+  solutionMapAfter = {}
+  for [size, [index, _]] in data[7]:
+    solutionMapBefore[tuple(size)] = [s["SolutionNameMin"] for s in data[5] if s["SolutionIndex"]==index][0]
+  for [size, [index, _]] in dataFiltered[7]:
+    solutionMapAfter[tuple(size)] = [s["SolutionNameMin"] for s in dataFiltered[5] if s["SolutionIndex"]==index][0]
+
+  assert solutionMapBefore == solutionMapAfter
+
+  # print final yaml for visual inspection
+  # stream = io.StringIO("")
+  # yaml.safe_dump(dataFiltered, stream, default_flow_style=None)
+  # print(stream.getvalue())
+
+def test_mergeLogic():
+  baseData = yaml.load(baseLogic, yaml.SafeLoader)
+  incData = yaml.load(incLogic, yaml.SafeLoader)
+
+  mergedData, *stats = mergeLogic(baseData, incData, False)
+
+  # check if stats are as expected
+  assert stats == [1,2,2] # 1 sizes added, 2 kernels added, 2 kernels removed/replaced
+
+  # print final yaml for visual inspection
+  # stream = io.StringIO("")
+  # yaml.safe_dump(mergedData, stream, default_flow_style=None)
+  # print(stream.getvalue())
+
+  # check if solution matches expected. assumes SolutionNameMin is uniqueSolution
+  # (which is satisfied by the test data given here)
+  solutionMap = {} # size -> solutionName
+  for size, [index, _] in mergedData[7]:
+    solutionMap[tuple(size)] = [s["SolutionNameMin"] for s in mergedData[5] if s["SolutionIndex"]==index][0]
+
+  sizeList = [tuple([256, 256, 1, 256]), tuple([128, 128, 1, 128]), tuple([64, 64, 1, 64])]
+  expectedSolution = [ "InUseForSize256or1024xxx", "InUseForSize128xxx", "InUseForSize128or64" ]
+  for size, expected in zip(sizeList, expectedSolution):
+    assert solutionMap[size] == expected
```

Comment:
Also check if tuple([1024, 1024 , 1, 1024]) maps to “InUseForSize256or1024xxx” ?

Reply:
Good catch